pkg/binding/interfaces.go

```diff
@@ -34,6 +49,14 @@ import (
 // try each method of interest and fall back to Event(EventEncoder) if none are supported.
 //
 type Message interface {
+	// TODO(slinkydeveloper) add docs
+	GetParent() Message
```

Comment:
This feels two things:

1 - I think GetParent is a little confusing, and I would expect this to be just GetMessage or GetWrappedMessage

2 - Consider breaking this out of the Message interface into a new WrappedMessage interface you can use to implement accessor methods and do a cast to WrappedMessage when you need to do the internal grab.

Reply:
Does it looks better now? https://github.com/cloudevents/sdk-go/pull/302/commits/259f6f4857e2136bfb9370459fb4f9bfb295e663

Reply:
+1 nice